### PR TITLE
CCP-415: add validation of minimum order amount to getCart calls

### PR DESCRIPTION
**Pull request by andre-kraus**

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Helper/Api2/Quote.php [hunk 1]
```diff
@@ -44,13 +45,45 @@ public function addItemErrors(Mage_Sales_Model_Quote $quote, Mage_Core_Model_Sto
...
+     * @throws Mage_Core_Model_Store_Exception
+     */
+    public function addQuoteErrors(Mage_Sales_Model_Quote $quote, Mage_Core_Model_Store $store)
+    {
+        $adminStore = Mage::app()->getStore();
+        Mage::app()->setCurrentStore($store->getCode());
+        /** @var Mage_Sales_Model_Quote_Item $item */
+        if (!$quote->validateMinimumAmount()) {
+            $minimumAmount = Mage::app()->getLocale()->currency($store->getCurrentCurrencyCode())
+                ->toCurrency(Mage::getStoreConfig('sales/minimum_order/amount'));
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Helper/Api2/Quote.php]: throws an exception is value is empty, the validateMinimum should prevent this from happening I am assuming. My mage is currently broken so I cannot check. In any case, at least the throws tag should be annotated

**Comment by augsteyer** [comment]
hmmm, the test you added is failing. I will need to fix my instance to check out why.

File: tests/postman/newman/collection.json [hunk 2]
```diff
@@ -2056,6 +2056,181 @@
...
+                              "});",
+                              "",
+                              "pm.test(\"Qty is indeed 1\", function () {",
+                              "    pm.expect(jsonData.items[0].qty).to.eql(1);",
+                              "});",
+                              "",
+                              "pm.test(\"Cart has errors\", function () {",
+                              "    pm.expect(jsonData.has_error).to.eql(true);",
+                              "    pm.expect(jsonData.errors).to.not.be.empty;",
+                              "});0"
```
Comment (augsteyer) [inline tests/postman/newman/collection.json]: there is this 0 at the end that needs cleaning up

**Comment by augsteyer** [comment]
Alrighty, I ran the tests using the same way it was to set it up via travis (using n98) and it fails the same way. It's a bit late over here to xDebug the cause, but I am assuming the issue is with store switching. The store in the validateMinimum thing that is used it picked up from the Quote (not global state), so you will need to double check what the store is on the quote and if it's set correctly:
screen shot 2018-04-25 at 21 00 16

File: tests/postman/newman/collection.json (hunk 2 above)
Comment (andre-kraus) [reply tests/postman/newman/collection.json]: removed it

File: .travis.yml [hunk 3]
```diff
@@ -67,6 +67,8 @@ env:
     CFG_API_CUSTOMER_NUMBER="11223344"
     CFG_API_SHOP_NUMBER="12345"
     CFG_API_KEY="11111111111111111111"
+    CFG_MINIMUM_ORDER_AMOUNT="21"
```
Comment (augsteyer) [inline .travis.yml]: you can remove this config from travis & magento_setup.sh, just hardcode in .magerun file.

**Comment by augsteyer** [comment]
as soon as the test clear, you can merge :)

**Pull request merged**